### PR TITLE
Add a new faction: the Norse Evangelicals

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Adds a new faction, a base for them, and several quests."

#### Purpose of change
Adds a new faction: the Norse Evangelicals. They are based in an old re-enactment set for Viking demonstrations, which they have fortified and made into a home. They are a blood-and-honor cult, mainly based around the xenophobic extermination of zombies and otherworldly creatures. They hold a special hatred for the Mycus and its disciples, considering them traitors to humanity. 

#### Describe the solution
The layout of their home fortress holds a henge in the middle, surrounded by a storehouse, a central longhouse containing the chief, homes, gardens, and mechanic shops. The fortress was intended to be a re-enactment site equipped with some modern amenities for the on-site staff. It was never finished before the Cataclysm; while the plumbing works, it has no electricity. 

The high gode (main leader) gives out combat-oriented quests centered around killing the Mycus, mi-gos, zombies, etc. 
The gode found at the henge offers more NPC-centric quests, like saving a raiding party or killing bandits.
The warehouse accountant is the local merchant, allowing players to buy bulk household items, medieval armor and weapons, and some assortment of guns. They offer far tamer missions than the leader; currently extending to fetch quests that open up bulk trade and loan opportunities later. 
The mechanics sell vehicles; specifically bikes, and provide basic fetch quests for automotive supplies.

#### Describe alternatives you've considered
I have considered rewriting the leaders' backstory to have the entire faction spawn on a delay, but I don't know how to do that. 

#### Testing
All dialogue options, quests, and trades work. No errors are given on loading a game with this content active.

#### Additional context
For some historical background, a 'gode' is a religious leader in the old Norse religions prior to the arrival of Christianity. The 'high gode' is of my own invention, seeing as how the current leader was a close friend of the cults' founder. "Reisendre" should mean "traveler" in Norse.
(SPOILER) -> The founder died to a hulk, revealed in the custom leader backstory.

If anyone knows otherwise, please correct me.

Current development plan:

- [x] Give the gode some quests
- [x] Give mechanics some jobs to do
- [x] Expand interactions with generic members
- [x] Expand quests for the high gode
- [x] Expand quests for the warehouse manager
- [x] Test code
- [x] Bugfix and lint
- [x] Finalize and un-draft for submission

### CODE REVIEW

File: data/json/npcs/valhalla_cult/TALK_valhallist_mechanic.json [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    "dynamic_line": "Sure, I'll get something for you.  I specialize in bikes of any sort, and I'll have it to you in reasonable time; all payments are upfront in signed chit.  So, what would you like?",
+    "responses": [
+      { 
+       "text": "[200$]  I'll take a bicycle.",
+       "condition": { "u_has_items": { "item": "signed_chit", "count": 80 } },
+       "effect": [ { "npc_add_var": "npc_making_car", "type": "timer", "context": "trade", "time": true }, { "npc_add_var": "currently_busy", "type": "general", "context": "trade", "value": "yes" }, { "npc_add_var": "u_want_bicycle", "type": "general", "context": "trade", "value": "yes" }, { "u_sell_item": "signed_chit", "count": 80 } ], 
+       "topic": "TALK_DONE" 
+      },
+      { 
+       "text": "[400$]  I'll take a mountian bike.",
```
Comment (TheShadowFerret) [inline data/json/npcs/valhalla_cult/TALK_valhallist_mechanic.json]:
```suggestion
       "text": "[400$]  I'll take a mountain bike.",

```

File: data/json/npcs/valhalla_cult/TALK_valhallist_mechanic.json [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+       "condition": { "u_has_items": { "item": "signed_chit", "count": 160 } },
+       "effect": [ { "npc_add_var": "npc_making_car", "type": "timer", "context": "trade", "time": true }, { "npc_add_var": "u_want_bicycle_dirt", "type": "general", "context": "trade", "value": "yes" }, { "npc_add_var": "currently_busy", "type": "general", "context": "trade", "value": "yes" }, { "u_sell_item": "signed_chit", "count": 160 } ], 
+       "topic": "TALK_DONE" 
+      },
+      { 
+       "text": "[550$]  I'll take an eletric bike.",
```
Comment (TheShadowFerret) [inline data/json/npcs/valhalla_cult/TALK_valhallist_mechanic.json]:
```suggestion
       "text": "[550$]  I'll take an electric bike.",

```

**Comment by MNG-cataclysm** [comment]
Fixed spelling issues given by the AI spellchecker, excluding nouns.

File: data/json/npcs/valhalla_cult/TALK_valhallist_shopkeeper.json [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        "condition": { "not": { "npc_has_effect": "asked_to_socialize" } },
+        "topic": "TALK_FRIEND_CHAT" 
+      },
+      { "text": "A pleasure chatting.  Gotta go!", "topic": "TALK_DONE" }
+    ]
+  },
+{
+    "type": "talk_topic",
+    "id": "TALK_VALHALLIST_SHOPKEEPER_WHO",
+    "dynamic_line": "I'm the warehouse manager, I just keep track of what we have at the moment.  Its not as exciting as what the warriors do, but its a necessary job.",
```
Comment (TheShadowFerret) [inline data/json/npcs/valhalla_cult/TALK_valhallist_shopkeeper.json:66]:
```suggestion
    "dynamic_line": "I'm the warehouse manager, I just keep track of what we have at the moment.  It's not as exciting as what the warriors do, but it's a necessary job.",

```

File: data/json/npcs/valhalla_cult/TALK_valhallist_shopkeeper.json [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    "responses": [
+      { "text": "Who are you?", "topic": "TALK_VALHALLIST_SHOPKEEPER_WHO" },
+      { "text": "Lets talk buisness, then.", "effect": "start_trade", "topic": "TALK_VALHALLIST_SHOPKEEPER_2" },
+      { "text": "That's cool, but I gotta got.", "topic": "TALK_DONE" }
+    ]
+  },
+{
+    "type": "talk_topic",
+    "id": "TALK_VALHALLIST_SHOPKEEPER_Loan_Hazmat",
+    "dynamic_line": "Certianly <name_g>, they've been a real help for us recently.  I can only spare one at a time though, so make sure to bring it back.",
```
Comment (TheShadowFerret) [inline data/json/npcs/valhalla_cult/TALK_valhallist_shopkeeper.json:85]:
```suggestion
    "dynamic_line": "Certainly <name_g>, they've been a real help for us recently.  I can only spare one at a time though, so make sure to bring it back.",

```

File: data/json/npcs/valhalla_cult/missions.json [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    "origins": [ "ORIGIN_OPENER_NPC" ],
+    "followup": "MISSION_VALHALLISTS_KILL_300_Z",
+    "dialogue": {
+      "describe": "Destroy a fungal bloom for the Valhallists.",
+      "offer": "My scouts have reported activity of those <mycus_slur>s nearby.  I want you to go over there and kill all the <mycus_slur> you see, then burn their colony to ash.  I have some Molotovs ready for you to do that with.",
+      "accepted": "May Thor protect you.",
+      "rejected": "They'll pay with their lives eventually.",
+      "advice": "Wear a gas mask and a hazard suit, which I wish we had to lend you.  Walking in those spore clouds can get you infected with them, and I'd rather not have to send an assassin to incinerate you.",
+      "inquire": "I don't see any ash on you.",
+      "success": "Those aliens had it coming to them.  You have done Odins' will today.",
```
Comment (TheShadowFerret) [inline data/json/npcs/valhalla_cult/missions.json:66]:
```suggestion
      "success": "Those aliens had it coming to them.  You have done Odin's will today.",

```

File: data/json/npcs/valhalla_cult/missions.json [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+    "origins": [ "ORIGIN_OPENER_NPC" ],
+    "dialogue": {
+      "describe": "Clear out a mi-go tower for the Valhallists.",
+      "offer": "We've found more aliens in the area.  My scouts have reported a scout tower for those weird multi-limbed aliens showed up recently, and that they might have prisoners inside.  I want you to go over there and kill every last <name_b> you can find, and free those prisoners.  I'm not looking to recruit anyone you free from there, just make sure they're alright.  Come back to me when you're done.",
+      "accepted": "May Thor protect you.",
+      "rejected": "They'll pay with their lives eventually, why wait?.",
+      "advice": "That tower is like a furnace on the interior, the scouts could even feel it from the outside.  They've got to have some source of heat though, so I'd destroy whatever's belching out that heat.  Also, shoot those <name_b>s from a distance, they'll fuck you up in hand-to-hand fighting.",
+      "inquire": "You're pretty clean for someone who should be killing <name_b>s.",
+      "success": "Those aliens had it coming to them.  You have done Odins' will today.",
```
Comment (TheShadowFerret) [inline data/json/npcs/valhalla_cult/missions.json:124]:
```suggestion
      "success": "Those aliens had it coming to them.  You have done Odin's will today.",

```

File: data/json/npcs/valhalla_cult/valhallist_leader_backstory.json [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      { "text": "How did you come to power here?", "topic": "TALK_VALHALLIST_LEADER_Position" },
+      { "text": "When did you build this place?", "topic": "TALK_VALHALLIST_LEADER_Compound" },
+      { "text": "<done_conversation_section>", "topic": "TALK_VALHALLIST_LEADER_2" },
+      { "text": "<end_talking>", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_VALHALLIST_LEADER_Ex_Job",
+    "dynamic_line": "I was a big history buff back then, but I worked in dead-end accounting for a university.  I had a three friends who I hung with after-hours; we were out on the town when <the_cataclysm> came.",
```
Comment (TheShadowFerret) [inline data/json/npcs/valhalla_cult/valhallist_leader_backstory.json:17]:
```suggestion
    "dynamic_line": "I was a big history buff back then, but I worked in dead-end accounting for a university.  I had three friends who I hung with after-hours; we were out on the town when <the_cataclysm> came.",

```

File: data/json/npcs/valhalla_cult/valhallist_leader_backstory.json [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    "responses": [
+      { "text": "When did you build this place?", "topic": "TALK_VALHALLIST_LEADER_Compound" },
+      { "text": "<done_conversation_section>", "topic": "TALK_VALHALLIST_LEADER_2" },
+      { "text": "<end_talking>", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_VALHALLIST_LEADER_Compound",
+    "dynamic_line": "One of my buddies wanted to get in the re-enacting buisness, so they started setting up this place as a to-scale model of what a Norse fort might have looked like.  I was planning on getting in to it with them, but <the_cataclysm> put a stop to it.  We came here from town, found the place wasn't occupied by those horrors, and set up camp here.  The Norse-obsessed guy decided to make a full religion out of it, and managed to bring back a large group fleeing one of those FEMA camps.",
```
Comment (TheShadowFerret) [inline data/json/npcs/valhalla_cult/valhallist_leader_backstory.json:27]:
```suggestion
    "dynamic_line": "One of my buddies wanted to get in the re-enacting business, so they started setting up this place as a to-scale model of what a Norse fort might have looked like.  I was planning on getting into it with them, but <the_cataclysm> put a stop to it.  We came here from town, found the place wasn't occupied by those horrors, and set up camp here.  The Norse-obsessed guy decided to make a full religion out of it, and managed to bring back a large group fleeing one of those FEMA camps.",

```

**Comment by MNG-cataclysm** [comment]
@dseguin Would you be willing to review this pull request, please?

**Comment by Maleclypse** [comment]
> 

You'll need to add all these words to the dictionary file.

**Comment by MNG-cataclysm** [comment]
> > 
> 
> You'll need to add all these words to the dictionary file.

The dictionary has been updated, all new terms are included in alphabetical order.

**Comment by MNG-cataclysm** [comment]
> > 
> 
> You'll need to add all these words to the dictionary file.

@Maleclypse The dictionary has been updated, all new terms are included in alphabetical order.

File: data/json/npcs/valhalla_cult/TALK_valhallist_gode.json [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    "responses": [ { "text": "Thanks.", "topic": "TALK_DONE" } ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_VALHALLIST_GODE_Religion",
+    "dynamic_line": "It seems likely you wouldn't know.  Before there was anything in the world, there was fires of Muspelhiem and the frigid ice of Niflheim.  When these two came together, the ice melted and became the first of the giants, Ymir.  As he slept, more giants sprung from his sweat and saliva.  From the melt came the cow Audhumla, who licked away the ice until she uncovered Buri, the first of the gods.  His children were Odin, Vili, and Ve; who slew Ymir to fashion the world from him.  They made the first people, Ask and Embla.  Odin fashioned all things which we see, but the destructive giants opposed and fought him; the hideous blob which invaded the earth came from the giants.  As such, we are dedicated to its extermination, to drive it from the face of the Earth until we see it no more.  Only then shall we undo <the_cataclysm>, and fashion the world anew.",
```
Comment (Maleclypse) [inline data/json/npcs/valhalla_cult/TALK_valhallist_gode.json:77]:
```suggestion
    "dynamic_line": "It seems likely you wouldn't know.  Before there was anything in the world, there was fires of Muspelhiem and the frigid ice of Niflheim.  When these two came together, the ice melted and became the first of the giants, Ymir.  As he slept, more giants sprung from his sweat and saliva.  From the melt came the cow Audhumla, who licked away the ice until she uncovered Buri, the first of the gods.  His children were Odin, Vili, and Ve; who slew Ymir to fashion the world from him.  They made the first people, Ask and Embla.  Odin fashioned all things which we see, but the destructive giants opposed and fought him; these undead which invaded the earth came from Hel as an army of the dead during Ragnarok.  As foretold in Vafthrúdnismál, Vidar and some of the younger generation of gods will create a new world.  Only then shall we undo <the_cataclysm>, and fashion the world anew.",
```
FYI, the blob doesn't exist as far as any survivors are concerned, not even Xedra knows about "the blob". Therefore any content mentioning "the blob" is unmergeable.

**Review by Maleclypse — APPROVED** [review]
Fairly solid.  Made some different choices than I would have in a couple places but reasonable.

File: data/json/npcs/valhalla_cult/TALK_valhallist_mechanic.json [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+    "dynamic_line": "That purpose is to serve the true lords of this world, and enter Valhalla on some coming day.  There I will not worry about the wandering dead, but see the might of battle and feast with those worthy.  Odin the Allfather will smile on me that day.",
+    "responses": [
+      { "text": "Let's talk about something else.", "topic": "TALK_VALHALLIST_MECHANIC_2" },
+      { "text": "Good luck with that.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_VALHALLIST_MECHANIC_2",
+    "dynamic_line": "Here to do buisness?",
```
Comment (Maleclypse) [inline data/json/npcs/valhalla_cult/TALK_valhallist_mechanic.json:34]:
```suggestion
    "dynamic_line": "Here to do business?",
```

File: data/json/npcs/valhalla_cult/TALK_valhallist_shopkeeper.json [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+      { "text": "That's cool, but I gotta got.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_VALHALLIST_SHOPKEEPER_Job",
+    "dynamic_line": "This is our warehouse, which I run and stock.  We've got everything here from our scavenging runs; books, food, tools, materials, you name it.  I'd be willing to sell a few things for the right price.",
+    "responses": [
+      { "text": "Who are you?", "topic": "TALK_VALHALLIST_SHOPKEEPER_WHO" },
+      { "text": "Lets talk buisness, then.", "effect": "start_trade", "topic": "TALK_VALHALLIST_SHOPKEEPER_2" },
```
Comment (Maleclypse) [inline data/json/npcs/valhalla_cult/TALK_valhallist_shopkeeper.json:90]:
```suggestion
      { "text": "Lets talk business, then.", "effect": "start_trade", "topic": "TALK_VALHALLIST_SHOPKEEPER_2" },
```

File: data/json/npcs/valhalla_cult/missions.json [hunk 12]
```diff
@@ -0,0 +1,515 @@
...
+    "item": "trade_writ",
+    "count": 1,
+    "difficulty": 30,
+    "value": 20000,
+    "origins": [ "ORIGIN_OPENER_NPC" ],
+    "start": { "assign_mission_target": { "om_terrain": "evac_center_18", "om_special": "evac_center", "reveal_radius": 0 } },
+    "has_generic_rewards": true,
+    "dialogue": {
+      "describe": "Go negotiate a trade agreement.",
+      "offer": "One of our raiding parties was out the other day, and came across an old evac shelter.  They managed to pull and adress for one of the refugee centers off the computer, and I mapped it out.  I figure they could make good trade partners, so I want you to go over there and work out a deal for me.",
```
Comment (Maleclypse) [inline data/json/npcs/valhalla_cult/missions.json:247]:
```suggestion
      "offer": "One of our raiding parties was out the other day, and came across an old evac shelter.  They managed to pull and address for one of the refugee centers off the computer, and I mapped it out.  I figure they could make good trade partners, so I want you to go over there and work out a deal for me.",
```

File: data/json/npcs/valhalla_cult/valhallist_leader_backstory.json [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
```
Comment (Maleclypse) [inline data/json/npcs/valhalla_cult/valhallist_leader_backstory.json:27]:
```suggestion
    "dynamic_line": "One of my buddies wanted to get in the re-enacting business, so they started setting up this place as a to-scale model of what a Norse fort might have looked like.  I was planning on getting in to it with them, but <the_cataclysm> put a stop to it.  We came here from town, found the place wasn't occupied by those horrors, and set up camp here.  The Norse-obsessed guy decided to make a full religion out of it, and managed to bring back a large group fleeing one of those FEMA camps.",
```

File: data/json/npcs/valhalla_cult/missions.json [hunk 14]
```diff
@@ -0,0 +1,515 @@
...
+    "monster_kill_goal": 1000,
+    "difficulty": 7,
+    "value": 0,
+    "origins": [ "ORIGIN_SECONDARY" ],
+    "followup": "MISSION_VALHALLISTS_KILL_MIGOS",
+    "end": { "effect": [ { "u_spawn_item": "signed_chit", "count": 500 } ] },
+    "has_generic_rewards": false,
+    "dialogue": {
+      "describe": "Kill 1000 zombies for the Valhallists",
+      "offer": "With the recent sucess we've had in killing the otherworldly threats, we must take grander strides in making the world safe again.  I need you to go out and kill one thousand <zombies> to do your new part in this endeavor.  A handsome reward will be provided upon your return.",
```
Comment (Maleclypse) [inline data/json/npcs/valhalla_cult/missions.json:147]:
```suggestion
      "offer": "With the recent success we've had in killing the otherworldly threats, we must take grander strides in making the world safe again.  I need you to go out and kill one thousand <zombies> to do your new part in this endeavor.  A handsome reward will be provided upon your return.",
```

File: data/json/npcs/valhalla_cult/missions.json [hunk 15]
```diff
@@ -0,0 +1,515 @@
...
+    "end": { "effect": [ { "u_spawn_item": "signed_chit", "count": 100 } ] },
+    "has_generic_rewards": false,
+    "dialogue": {
+      "describe": "Kill 300 zombies for the Valhallists",
+      "offer": "You've proven yourself as a competent warrior, and we need your services again.  Scavenging runs have been delayed recently due to hordes of undead blocking off towns, and nearly killing one of us last time.  I need you to help exterminate this threat, so go kill three hundred of those <name_b>s for us.",
+      "accepted": "May Thor protect you.",
+      "rejected": "We are going to have to face them sometime, now is better than ever.",
+      "advice": "Slay a few at a time, and don't let them surround you.",
+      "inquire": "Are the dead laid to rest?",
+      "success": "We shall see a great sucess in the future now that our supply lines are open again.  Gudene vil det!",
```
Comment (Maleclypse) [inline data/json/npcs/valhalla_cult/missions.json:92]:
```suggestion
      "success": "We shall see a great success in the future now that our supply lines are open again.  Gudene vil det!",
```

**Comment by bombasticSlacks** [comment]
Gotta fix the actual typos the spell checker spotted.

File: data/json/npcs/valhalla_cult/missions.json [hunk 16]
```diff
@@ -0,0 +1,515 @@
...
+    "origins": [ "ORIGIN_OPENER_NPC" ],
+    "has_generic_rewards": false,
+    "dialogue": {
+      "describe": "Emancipate slaves for the Valhallists.",
+      "offer": "We've tracked some bandits back to one of their camps, and our scouts have reported that slaves are being held there.  I need you to destroy the camp and free those slaves, then bring them back here.  We'll give them room and board, and hopefully they'll join us.",
+      "accepted": "May Thor protect you.",
+      "rejected": "Are you certain?  It'll be worth your while.",
+      "advice": "I'd ambush the bandits at night, if it were me.  They've got the perimeter trapped with mines, too.",
+      "inquire": "Where are they?",
+      "success": "The world will be a better place without theives walking its surface.",
```
Comment (Maleclypse) [inline data/json/npcs/valhalla_cult/missions.json:433]:
```suggestion
      "success": "The world will be a better place without thieves walking its surface.",
```

**Comment by Maleclypse** [comment]
> Gotta fix the actual typos the spell checker spotted.

Only theives is a typo unless I'm mistaken out of what's left in the spell checker.

**Pull request merged**